### PR TITLE
Beth/issue 625

### PR DESCRIPTION
**Pull request by bhg**
Fix for [issue 625](https://github.com/DemocracyLab/CivicTechExchange/issues/625)

### CODE REVIEW

File: civictechprojects/admin.py [hunk 1]
```diff
@@ -22,13 +26,21 @@ class GroupAdmin(admin.ModelAdmin):
     list_display = tuple(group_text_fields) + ('group_creator',) + group_filter_fields
     search_fields = group_text_fields + ['group_creator__email']
     list_filter = group_filter_fields
+    def save_model(self, request, obj, form, change):
+        if change:
+            obj.recache()
```
Comment (marlonkeating) [inline civictechprojects/admin.py:31]: Add update_linked_items

File: civictechprojects/admin.py [hunk 2]
```diff
@@ -9,6 +9,10 @@ class ProjectAdmin(admin.ModelAdmin):
     list_display =  tuple(project_text_fields) + ('project_creator',) + project_filter_fields
     search_fields = project_text_fields + ['project_creator__email']
     list_filter = project_filter_fields
+    def save_model(self, request, obj, form, change):
+        if change:
+            obj.recache(recache_linked=False)
```
Comment (marlonkeating) [inline civictechprojects/admin.py:14]:
```suggestion
            obj.recache(recache_linked=True)
```

File: civictechprojects/admin.py [hunk 3]
```diff
@@ -22,13 +26,21 @@ class GroupAdmin(admin.ModelAdmin):
...
 
 event_text_fields = ['event_name', 'event_agenda', 'event_description','event_location', 'event_rsvp_url', 'event_live_id', 'event_short_description']
 event_filter_fields = ('event_date_created', 'event_date_end', 'event_date_modified', 'event_date_start', 'is_searchable', 'is_created')
 class EventAdmin(admin.ModelAdmin):
     list_display = tuple(event_text_fields) + ('event_creator',) + event_filter_fields
     search_fields = event_text_fields + ['event_creator__email']
     list_filter = event_filter_fields
+    def save_model(self, request, obj, form, change):
+        if change:
+            obj.recache()
```
Comment (marlonkeating) [inline civictechprojects/admin.py:42]: Add update_linked_items

**Review by marlonkeating — CHANGES_REQUESTED** [review]

File: civictechprojects/admin.py [hunk 4]
```diff
@@ -22,13 +26,23 @@ class GroupAdmin(admin.ModelAdmin):
     list_display = tuple(group_text_fields) + ('group_creator',) + group_filter_fields
     search_fields = group_text_fields + ['group_creator__email']
     list_filter = group_filter_fields
+    def save_model(self, request, obj, form, change):
+        if change:
+            obj.recache()
+            obj.update_linked_items()
```
[the remark this answers is not in the document]
Comment (bhg) [reply civictechprojects/admin.py:33]: Done

File: civictechprojects/admin.py [hunk 5]
```diff
@@ -9,6 +9,10 @@ class ProjectAdmin(admin.ModelAdmin):
     list_display =  tuple(project_text_fields) + ('project_creator',) + project_filter_fields
     search_fields = project_text_fields + ['project_creator__email']
     list_filter = project_filter_fields
+    def save_model(self, request, obj, form, change):
+        if change:
+            obj.recache(recache_linked=True)
+        super().save_model(request, obj, form, change)
```
Comment (marlonkeating) [inline civictechprojects/admin.py:15]: This recaching seems to be working even though the recache is done before the save operation, but let's put the operations in the save -> recache order for consistency if nothing else.

**Review by marlonkeating — CHANGES_REQUESTED** [review]

File: civictechprojects/admin.py (hunk 5 above)
Comment (bhg) [reply civictechprojects/admin.py:15]: Not sure how I missed that one. Fixing.

**Review by marlonkeating — APPROVED** [review]

**Pull request merged**